bootstrap.go

```diff
@@ -52,9 +59,9 @@ func initTracer(cfg config.DatadogParameters, isConnectionSocket bool) {
 	tracer.Start(tracerOptions...)
 }
 
-func initProfiler(cfg config.DatadogParameters, enableExtraProfiling bool, isConnectionSocket bool) error {
+func initProfiler(cfg config.DatadogParameters, connectionType ConnectionType) error {
 	var profilerTypes []profiler.ProfileType
-	if enableExtraProfiling {
+	if cfg.GetEnableExtraProfiling() {
 		profilerTypes = []profiler.ProfileType{
```

Comment:
@nhhagen Was referring to this, if you would like to make it nice and flexible for consumer, then:

Like options would be nice
opts := []DatadogProfilers{profiler.CPUProfile, profiler.HeapProfile} etc
AddProfiling(opts...)

Or even more clean way would be "builder pattern" where you can call Option as Method and then build a slice in the end.

Reply:
Agreed